dealer/dealer.go

```diff
@@ -185,16 +212,16 @@ loop:
 		}
 	}
 
-	_ = d.conn.Close(websocket.StatusInternalError, "")
-
 	// if we shouldn't stop, try to reconnect
 	if !d.stop {
-		d.reconnectLock.Lock()
+		_ = d.conn.Close(websocket.StatusInternalError, "")
```

Comment:
Why do this only when not stopping?

Reply:
Because `dealer.Close()` handles closing the connection when we are stopping